Dockerfile2

```diff
@@ -0,0 +1,164 @@
+FROM ubuntu:focal as minimal-system
```

Comment:
Dockerfile2 -- Accidentally checked in?

Reply:
Dockerfile2 was written to do the same thing as the Dockerfile but differently in case of the development stage in order to reduce image size and keeping it DRY.

With `Dockerfile` I have to use additional intermediate stages to [install](https://github.com/openedx/edx-platform/blob/56afc15a8162f5ad3068db920b045b150c91bae8/Dockerfile#L124-L126) and [COPY](https://github.com/openedx/edx-platform/blob/56afc15a8162f5ad3068db920b045b150c91bae8/Dockerfile#L150) over the development environment dependencies (in addition to the production ones being [copied](https://github.com/openedx/edx-platform/blob/56afc15a8162f5ad3068db920b045b150c91bae8/Dockerfile#L132), both set of dependencies are necessary for the development environment to function properly). This increases the image size for development environment to 3.06 GB due to 2 layers of COPY instructions which are copying the same directory, first with the production dependencies, and second with both production + development dependencies. Technically, I can rewrite this to ensure that 2 COPY instructions don't happen during the build process for the development image to reduce size. However that will end up with the Dockerfile having multiple COPY instructions in multiple stages for the same artifacts and it will not be DRY.

In `Dockerfile2` I added the `DEVELOPMENT` argument and set it to false by default, which will prevent installation of development dependencies for an image targeting `production` by default. If someone needs to build an image targeting `development`, they will have to set `--target development` and `--build-arg DEVELOPMENT=true` which will install dependencies for development environment. This not only keeps the development image size to 2.22 GB but also keeps the Dockerfile DRY by keeping just one COPY statement for the virtualenv.

I wanted some feedback on `Dockerfile2` which is why I commited it to the branch.